src/playlist.rs

```diff
@@ -977,8 +841,99 @@ mod test {
+),
+            ..Default::default()
+        };
+        let seventh_song = Song {
+            path: Path::new("path-to-seventh").to_path_buf(),
+            analysis: Analysis::new([0.3853314, -0.8475499, -0.64330614, -0.85917395, -0.6624141, -0.6356613, -0.40988427, -0.7480691, 0.45981812, 0.47096932, -0.19245929, -0.5228787, -0.42246288, -0.52656835, -0.45702273, -0.569838, -0.97620565, -0.97741324, -0.9776932, -0.98088175]
+),
+            ..Default::default()
+        };
+        let eight_song = Song {
+            path: Path::new("path-to-eight").to_path_buf(),
+            analysis: Analysis::new([0.18926656, -0.86667925, -0.7294189, -0.856192, -0.7180501, -0.66697484, -0.6093149, -0.82118326, 0.3888924, 0.42430043, -0.4414854, -0.6957753, -0.7092425, -0.68237424, -0.55543846, -0.77678657, -0.98610276, -0.98707336, -0.99165493, -0.99011236]
+),
+            ..Default::default()
+        };
+        let ninth_song = Song {
+            path: Path::new("path-to-ninth").to_path_buf(),
+            analysis: Analysis::new(
+                [0.38328362, -0.8752751, -0.8165319, -0.948534, -0.77668643, -0.9051969, -0.8473458, -0.88643366, 0.49641085, 0.5132351, -0.41367024, -0.5279201, -0.46787983, -0.49218357, -0.42164963, -0.6597451, -0.97317076, -0.9800342, -0.9832096, -0.98385316]
+),
+            ..Default::default()
+        };
+        let tenth_song = Song {
+            path: Path::new("path-to-tenth").to_path_buf(),
+            analysis: Analysis::new(
+                [0.4301988, -0.89864063, -0.84993315, -0.9518692, -0.8329567, -0.9293889, -0.8605237, -0.8901016, 0.35011983, 0.3822446, -0.6384951, -0.7537949, -0.5867439, -0.57371, -0.5662942, -0.76130676, -0.9845436, -0.9833387, -0.9902381, -0.9905396]
+),
+            ..Default::default()
+        };
+        let eleventh_song = Song {
+            path: Path::new("path-to-eleventh").to_path_buf(),
+            analysis: Analysis::new(
+                [0.42334664, -0.8632808, -0.80268145, -0.91918564, -0.7522441, -0.8721291, -0.81877685, -0.8166921, 0.53626525, 0.540933, -0.34771818, -0.45362264, -0.35523874, -0.4072432, -0.25506926, -0.553644, -0.9624399, -0.9706371, -0.9753268, -0.9764576]
+),
+            ..Default::default()
+        };
+
+        let mut songs = [
+            &tenth_song, &first_song, &third_song, &eleventh_song, &fifth_song, &second_song, &ninth_song, &sixth_song, &fourth_song, &eight_song, &seventh_song
+        ];
+
+        // We train the algorithm on the Mozart concertos, and the expectation is that the tracks
+        // from the Miles Davis record will end up last.
+        let opts = ForestOptions { n_trees: 1000, sample_size: 200, max_tree_depth: None, extension_level: 10 };
+        closest_to_songs(&[&first_song, &second_song, &third_song, &ninth_song, &tenth_song, &eleventh_song], &mut songs, &opts);
```

Comment:
could we have the `closest_to_songs` call without one of the Mozart tracks, to make sure that even if it's not trained on that specific song, it still ranks better than Miles'?

Don't sweat it if it doesn't work though, I think the test is still useful enough as it is :smile:

Reply:
Yep, that seems to work. I train on one of the concertos, and it still works for ~8k runs of the test here.